### PR TITLE
EVG-15534: allow plugin GraphQL support

### PR DESCRIPTION
**Pull request by hadjri**
[EVG-15534](https://jira.mongodb.org/browse/EVG-15534)

### Description 
Follow up from the following PR: https://github.com/evergreen-ci/evergreen/pull/5154
Modified graphql schema in accordance with new required plugin inputs and ran gqlgen to update generated.go to allow plugin GraphQL support
### Testing 
Added tests to project settings and repo settings resolvers with new updated schema.

### CODE REVIEW

**Comment by malikchaya2** [comment]
We conventionally also add the UI team to PRs that are focused on the graphql schema and resolvers, so I'm adding them as reviewers as well.

**Review by malikchaya2 — COMMENTED** [review]
It looks like this project also added PerfEnabled to the Project struct. Is that configurable by users? Should we be adding that here as well?

File: graphql/schema.graphql [hunk 1]
```diff
@@ -462,6 +464,8 @@ input RepoRefInput {
   filesIgnoredFromCache: [String!]
   disabledStatsCache: Boolean
   workstationConfig: WorkstationConfigInput
+  buildBaronSettings: BuildBaronSettingsInput
```
Comment (SupaJoon) [inline graphql/schema.graphql:467]: Do these additional inputs require changes to [SaveProjectSettingsForSection](https://github.com/evergreen-ci/evergreen/blob/main/rest/data/project_settings.go#L78) for them to processed as part of the saveRepoSettingsForSection mutation?

Comment (hadjri) [reply graphql/schema.graphql:467]: Yes, that has been done in this PR: https://github.com/evergreen-ci/evergreen/pull/5154

**Review by ablack12 — APPROVED** [review]

**Review by malikchaya2 — APPROVED** [review]

**Review by SupaJoon — APPROVED** [review]

File: graphql/schema.graphql [hunk 2]
```diff
@@ -498,6 +504,31 @@ input TaskSyncOptionsInput {
...
 }
 
+input BuildBaronSettingsInput {
+  ticketCreateProject: String!
+  ticketSearchProjects: [String!]
+  bfSuggestionServer: String!
+  bfSuggestionUsername: String!
+  bfSuggestionPassword: String!
+  bfSuggestionTimeoutSecs: Int!
+  bfSuggestionFeaturesURL: String!
```
Comment (khelif96) [inline graphql/schema.graphql:514]: Just to validate that you are aware of this and that you are intentionally adding the `!`. the `!` means that this field is required. And will not pass validation unless those fields are included in the request.  So a user would not be able to partially update one of these fields and would be required to pass in the full object when ever passing this input

File: graphql/schema.graphql [hunk 3]
```diff
@@ -498,6 +504,31 @@ input TaskSyncOptionsInput {
   patchEnabled: Boolean!
 }
 
+input BuildBaronSettingsInput {
+  ticketCreateProject: String!
+  ticketSearchProjects: [String!]
```
Comment (khelif96) [inline graphql/schema.graphql:509]: This line indicates this field can be null or an array with String elements. But the rest of the fields are all fully required

**Review by khelif96 — COMMENTED** [review]
Should there be a change in any of the resolvers to accommodate the changes in this pr? You will need to merge in main and rerun gqlgen if you haven't.

**Comment by sophstad** [comment]
Just wanted to flag this PR that makes some changes to the `SaveProjectSettingsForSection` and `SaveRepoSettingsForSection` resolvers; the main change required is that your new `ProjectPagePluginSection` const should use an uppercase string to align with the GraphQL enum: https://github.com/evergreen-ci/evergreen/pull/5186

File: graphql/schema.graphql [hunk 4]
```diff
@@ -1292,11 +1292,11 @@ type WorkstationConfig {
...
-  bfSuggestionUsername: String!
-  bfSuggestionPassword: String!
-  bfSuggestionTimeoutSecs: Int!
-  bfSuggestionFeaturesURL: String!
+  bfSuggestionServer: String
+  bfSuggestionUsername: String
+  bfSuggestionPassword: String
+  bfSuggestionTimeoutSecs: Int
+  bfSuggestionFeaturesURL: String
 }
```
Comment (sophstad) [inline graphql/schema.graphql:1301]: Sorry forgot to mention -- `PLUGINS` must also be added to the ProjectSettingsSection enum in this file. Thanks!

**Review by sophstad — CHANGES_REQUESTED** [review]

**Review by sophstad — APPROVED** [review]

**Pull request merged**